server/fileserver.go

```diff
@@ -101,7 +101,13 @@ func (fh *fileHandler) serveFile(w http.ResponseWriter, r *http.Request, name st
 	// Still a directory? (we didn't find an index file)
 	// Return 404 to hide the fact that the folder exists
 	if d.IsDir() {
-		return http.StatusNotFound, nil
+		if name == "/" {
```

Comment:
Well, to be fair, the server's name is already exposed in the Server field of the response headers.